src/field_msg.cpp

```diff
@@ -21,60 +23,60 @@ FieldMessage::FieldMessage()
     (*this)[e_fields::field_cnt] = {cast(e_fields::field_cnt), field_type::int32, true};
 }
 
-const FieldMessage::fld_el &FieldMessage::operator[](e_fields fld) const
+const Message::fld_el &Message::operator[](e_fields fld) const
 {
     return els[cast(fld)];
 }
 
-FieldMessage::fld_el &FieldMessage::operator[](e_fields fld)
+Message::fld_el &Message::operator[](e_fields fld)
 {
     return els[cast(fld)];
 }
 
 template <>
-void FieldMessage::set_bit<false>(int ind)
+void Message::set_bit<false>(int ind)
 {
     msg_bitmap &= ~(static_cast<int64_t>(1) << ind);
 }
 
 template <>
-void FieldMessage::set_bit<true>(int ind)
+void Message::set_bit<true>(int ind)
 {
     msg_bitmap |= static_cast<int64_t>(1) << ind;
 }
 
-int FieldMessage::cast(e_fields fld) const
+int Message::cast(e_fields fld)
 {
     return static_cast<int>(fld);
 }
 
-void FieldMessage::_throw(const char *str)
+void Message::_throw(const char *str)
 {
     throw std::invalid_argument(str);
 }
 
-bool FieldMessage::is_constant(e_fields fld)
+bool Message::is_constant(e_fields fld)
 {
     return fld == e_fields::message_size ||
            fld == e_fields::corrId ||
            fld == e_fields::field_cnt;
 }
 
-void FieldMessage::throw_on_constant(e_fields fld)
+void Message::throw_on_constant(e_fields fld)
 {
     if (is_constant(fld))
         _throw("this value cant be modified");
 }
 
-size_t FieldMessage::fld_el::size()
+size_t Message::fld_el::size()
 {
     auto &curr = *this;
     if (curr.type == field_type::int32)
         return sizeof(int32_t);
     return std::get<std::string>(curr.value).size();
 }
 
-void FieldMessage::set(e_fields fld, const std::string &val)
+void Message::set(e_fields fld, const std::string &val)
```

Comment:
Move semantic is your friend
```suggestion
void Message::set(e_fields fld, std::string val)
```